### PR TITLE
Update contribution guide

### PR DESCRIPTION
**Pull request by jcorioland**
- Update the how to work with locally built provider section
- Add instruction on how to get started with developing the Azure RM provider, using the Azure SDK for Go etc...

### CODE REVIEW

**Comment by jcorioland** [comment]
@grayzu fyi

File: provider/CONTRIBUTE.md
Comment (grayzu) [inline provider/CONTRIBUTE.md]: Spelling: "... recommend that you ..."

File: provider/CONTRIBUTE.md
Comment (grayzu) [inline provider/CONTRIBUTE.md]: update wording to "... to make sure that the repo maintainers are aware that you are going to work on this issue." We should also have them submit the schema that they intend to implement so that feedback can be provided on that before it is implemented. This will save significant rework.

**Review by grayzu — CHANGES_REQUESTED** [review]
Few minor changes to wording.

@JunyiYi can you have a quick look to ensure that best practices are being suggested here.

**Comment by jcorioland** [comment]
@grayzu I've just done the update you have requested

File: provider/CONTRIBUTE.md
Comment (JunyiYi) [inline provider/CONTRIBUTE.md]: Example is optional. Typically for my contributions, I do not add examples.

File: provider/CONTRIBUTE.md
Comment (JunyiYi) [inline provider/CONTRIBUTE.md]:
```suggestion
### Developing a Data Source
```

File: provider/CONTRIBUTE.md
Comment (JunyiYi) [inline provider/CONTRIBUTE.md]: Data source schema is much simpler than resource. You only need to provide required properties to locate that resource (for example: [API Management data source](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/data_source_api_management.go#L18-L20)), and mark all other properties returned from the service as `Computed`. There are no `ForceNew`, no `MinItems`, no `MaxItems`, etc.

File: provider/CONTRIBUTE.md
Comment (JunyiYi) [inline provider/CONTRIBUTE.md]:
```suggestion
- One function to define the schema of the resource, named by convention resourceArm*NAME_OF_RESOURCE*, for example `resourceArmBatchAccount`.
```

**Review by JunyiYi — COMMENTED** [review]
Thanks @jcorioland for the contributions. The documentation looks good to me. I have provided some comments here, and it would be good to go when they are resolved.

File: provider/CONTRIBUTE.md
Comment (jcorioland) [reply provider/CONTRIBUTE.md]: Done.

File: provider/CONTRIBUTE.md
Comment (jcorioland) [reply provider/CONTRIBUTE.md]: Done.

File: provider/CONTRIBUTE.md
Comment (jcorioland) [reply provider/CONTRIBUTE.md]: Done.

File: provider/CONTRIBUTE.md
Comment (jcorioland) [reply provider/CONTRIBUTE.md]: Done.

**Comment by jcorioland** [comment]
Thanks @JunyiYi for the review. I have pushed updates according to your comments.

**Review by JunyiYi — APPROVED** [review]
Thanks @jcorioland for the update, approved.

**Review by grayzu — APPROVED** [review]

**Pull request merged**
